lib/matplotlib/image.py

```diff
@@ -368,8 +368,9 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                     # values to carry the over/under/bad information
                     rgba = np.empty((A.shape[0], A.shape[1], 4), dtype=A.dtype)
                     rgba[..., 0] = A  # normalized data
-                    rgba[..., 1] = A < 0  # under data
-                    rgba[..., 2] = A > 1  # over data
+                    with np.errstate(invalid='ignore'):
+                        rgba[..., 1] = A < 0  # under data
+                        rgba[..., 2] = A > 1  # over data
```

Comment:
I think this is hiding the real problem. `A` is the output of a norm, which ensures it is a masked array.  Otherwise, the line below would fail.  Therefore the problem is that the norm is not running its floating point input through `np.ma.masked_invalid` or equivalent, so the bad values are not getting masked, resulting in the warnings and in the line below being incorrect, because it is not finding all of the bad data.